src/logic/currencyValues/store/selectors/index.ts

```diff
@@ -1,46 +1,41 @@
-import { List, Map, RecordOf } from 'immutable'
 import { createSelector } from 'reselect'
 
-import { CURRENCY_VALUES_KEY } from 'src/logic/currencyValues/store/reducer/currencyValues'
-import { safeParamAddressFromStateSelector } from 'src/routes/safe/store/selectors'
-import { AppReduxState } from 'src/store/index'
 import {
-  AVAILABLE_CURRENCIES,
-  BalanceCurrencyRecord,
-  CurrencyRateValue,
-  CurrencyRateValueRecord,
-} from 'src/logic/currencyValues/store/model/currencyValues'
+  CURRENCY_VALUES_KEY,
+  CurrencyReducerMap,
+  CurrencyValuesState,
+} from 'src/logic/currencyValues/store/reducer/currencyValues'
+import { safeParamAddressFromStateSelector } from 'src/routes/safe/store/selectors'
+import { AppReduxState } from 'src/store'
+import { CurrencyRateValue } from 'src/logic/currencyValues/store/model/currencyValues'
 import { BigNumber } from 'bignumber.js'
 
-export const currencyValuesSelector = (state: AppReduxState): Map<string, RecordOf<CurrencyRateValue>> =>
-  state[CURRENCY_VALUES_KEY]
+export const currencyValuesSelector = (state: AppReduxState): CurrencyValuesState => state[CURRENCY_VALUES_KEY]
 
 export const safeFiatBalancesSelector = createSelector(
   currencyValuesSelector,
   safeParamAddressFromStateSelector,
-  (currencyValues, safeAddress): CurrencyRateValueRecord => {
+  (currencyValues, safeAddress): CurrencyReducerMap | undefined => {
     if (!currencyValues) return
     return currencyValues.get(safeAddress)
   },
 )
 
+const currencyValueSelector = <K extends keyof CurrencyRateValue>(key: K) => (
+  currencyValuesMap?: CurrencyReducerMap,
+): CurrencyRateValue[K] => currencyValuesMap?.get(key) || null
```

Comment:
is the return type ok with null being returned? also, you can use notSetValue param for `.get`, i've pointed that already a lot of times

Reply:
I get what you say about the `notSetValue`, will fix the `CurrencyReducerMap`, but won't use it here.

What will happen if `currencyValuesMap` is `undefined`? With your proposed solution this will be returning `undefined`, a `value` or `null`.

I'll remove `null` from the _equation_ here.

Reply:
why could it be undefined? we should have a default value for that (= empty map)

Reply:
if you check what was before:

```typescript
export const safeFiatBalancesListSelector = createSelector(safeFiatBalancesSelector, (currencyValuesMap) => {
  if (!currencyValuesMap) return
  return currencyValuesMap.get('currencyBalances') ? currencyValuesMap.get('currencyBalances') : List([])
})

export const currentCurrencySelector = createSelector(safeFiatBalancesSelector, (currencyValuesMap) =>
  currencyValuesMap ? currencyValuesMap.get('selectedCurrency') : null,
)

export const currencyRateSelector = createSelector(safeFiatBalancesSelector, (currencyValuesMap) =>
  currencyValuesMap ? currencyValuesMap.get('currencyRate') : null,
)
```

...we had a mixed type of returns values.

I think that this solution works fine for the three selectors and makes it less verbose and easy to reason about. That was my motivation, actually.